sig-cli/charter.md

```diff
@@ -5,25 +5,81 @@ the Roles and Organization Management outlined in [sig-governance].
+[design principles](design-principles.md) for the focus of kubectl functionality.
+
+This group focuses on command line tooling for working with Kubernetes APIs and Resource
+Config.  This includes both generalized tooling for working with Resources, Resource Config
+and Resource Types (e.g. using resource / object metadata, duck-typing, openapi, discovery, 
+scale and status subresources, etc) as well as tooling for working with specific Kubernetes
+APIs (e.g. `logs`, `exec`, `create configmap`).
+
+The scope includes both low level tooling that may be used by scripts or higher-level
+solutions, as well as higher level porcelain to reduce user friction for simple,
+common, difficult or important tasks performed by users (e.g. `edit`, `top`, `wait`).  The
+scope also includes publishing a subset the libraries which were used to develop the tooling
+itself.
+
+The decision whether to publish specific functionality as part of kubectl,
+as a separate tool, as a kubectl extension, or as a library are technical
+decisions made by the SIG and owners of the code under development.
+
+#### Examples of Functionality In Kubectl
+
+Following are examples of functionality that is in kubectl.
+
+- Invoking Kubernetes APIs:
+  - Resource APIs - e.g. `create`, `replace`, `delete`, `patch`, `get`, etc
+  - SubResource APIs - e.g. `exec`, `attach`, `logs`, `scale`, etc
+  - Discovery Service - e.g. `api-versions`, `api-resources`
+  - Version - e.g. `version`
+  - OpenAPI - e.g. `explain`
+- Pre and Post processing API Resource Config, API Requests and API Responses
+  - `apply` - creates a patch from Resource Config by performing a 3-way diff
+  - `get` - formats API responses into columns
+- Aggregating multiple API Responses and post processing them
+  - `describe` - performs multiple calls and aggregates results
+  - `run --expose` - creates multiple resources
+- Collapsing multiple manual steps into a command
+  - `edit` - collapses `get`, edit a file, `patch` into a single command
+  - `cp` - collapses manual steps for copy a file out of a container
+- Generating Kubernetes Resource Config locally or creating Resources remotely
+  - From commands + arguments + flags - e.g. `create configmap`, `run`, `expose`
+  - From declarative files - Kustomization `configMapGenerator`, `secretMapGenerator`
+- Transforming Kubernetes Resource Config locally or patching remotely
+  - From commands + arguments + flags - e.g. `annotate`, `set image`, `patch`, etc
+  - From declarative files - e.g. Kustomization `commonAnnotations`, `images`, `patches`
+  - From flags - e.g. `-n` - set the namespace of Resources
+- Blocking on propagation of an event or change to the cluster
+  - e.g. `wait`, `rollout`
+- Defining or referencing a collection of Resource Config
+  - e.g. `-f` can reference a url, a file containing multiple Resources, or a directory
+  - e.g. `-R` can traverse a directory recursively
+  - e.g. Kustomization `bases` can refer to other Kustomizations
```

Comment:
First reference to Kustomize as far as I can tell, perhaps delete, since a generic description would work.

Reply:
None of the commands listed here are documented here. Links to documentation could be inserted. Or this could be extracted into another doc. Or perhaps just summarized. 

No other SIG catalogs every API and feature in their charter. I don't think this level of specificity in a charter is a precedent we want to set.

Reply:
I am good with deleting them.